pilot/pkg/model/config.go

```diff
@@ -276,6 +277,14 @@ type IstioConfigStore interface {
 	// Mixerclient end user authn policy specifications associated
 	// with destination service instances.
 	EndUserAuthenticationPolicySpecByDestination(instance *ServiceInstance) []Config
+
+	// AuthenticationPolicyByDestination selects authentication policy associated
+	// with a service + port. Hostname must be FQDN.
+	// If there are more than one policies at different scopes (global, namespace, service)
+	// the one with the most specific scope will be selected. If there are more than
+	// one with the same scope, the first one seen will be used (later, we should
+	// have validation at submitting time to prevent this scenario from happening)
+	AuthenticationPolicyByDestination(hostname string, port *Port) *authn.Policy
```

Comment:
return value should be *Config

Reply:
Out of curiosity, what are the reasons for that interface, i.e return *Config? If caller just needs the spec data, doing that requires another call to config.Spec.(...) outside. I find that is a bit less elegant.

Reply:
I don't have an answer...

Reply:
So can we keep as this?

Reply:
`AuthenticationPolicyByDestination` should be consistent with all of the other config store accessors and use `*Config` or `[]Config`. Use of `Config` allows for callers to access object metadata if they need it. 

Ideally, we could refactor the object representation a bit so that the top-level types (e.g. authn.Policy) include the object metadata similar to k8s objects.

Reply:
Change to return *Config.

However, unless there are more compelling reasons, I still feel it's unnecessary to keep this consistency, if some callers do not care much for config metadata.